game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -54,7 +54,7 @@
   private boolean isDefensive;
   private boolean isBombing;
 
-  ProCombatMoveAi(final ProAi ai) {
+  ProCombatMoveAi(final AbstractProAi ai) {
```

Comment:
nit, we probably should be shooting for the dependency to be on an interface where it so happens that `AbstractProAi` implements that interface. To get there we may have to think a bit more about the modelling of this one AI and which interfaces it makes sense to have.

To another extent, we should perhaps also consider 'interface segregation principal' and how many interfaces we would need so that a dependency like this is only given an interface where it actually uses everything that is passed to it.